server/src/main/java/org/opensearch/action/search/SearchCoordinatorStats.java

```diff
@@ -0,0 +1,169 @@
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.common.metrics.MeanMetric;
+
+/**
+ * Coordinator level search stats
+ *
+ * @opensearch.internal
+ */
+public final class SearchCoordinatorStats implements SearchRequestOperationsListener {
+    public StatsHolder totalStats = new StatsHolder();
+
+    // private final CounterMetric openContexts = new CounterMetric();
+
+    // private volatile Map<String, StatsHolder> groupStats = emptyMap();
+    private long queryPhaseStart;
+    private long queryPhaseEnd;
+    private long fetchPhaseStart;
+    private long fetchPhaseEnd;
+    private long expandSearchPhaseStart;
+    private long expandSearchPhaseEnd;
+
+    public long getQueryPhaseStart() {
+        return queryPhaseStart;
+    }
+    public long getQueryPhaseEnd() {
+        return queryPhaseEnd;
+    }
+    public long getFetchPhaseStart() {
+        return fetchPhaseStart;
+    }
+    public long getFetchPhaseEnd() {
+        return fetchPhaseEnd;
+    }
+    public long getExpandSearchPhaseStart() {
+        return expandSearchPhaseStart;
+    }
+    public long getExpandSearchPhaseEnd() {
+        return expandSearchPhaseEnd;
+    }
+
+    @Inject
+    public SearchCoordinatorStats () { }
+    public long getQueryMetric() {
+        return totalStats.queryMetric.sum();
+    }
+    public long getQueryCurrent() {
+        return totalStats.queryCurrent.count();
+    }
+    public long getQueryTotal() {
+        return totalStats.queryTotal.count();
+    }
+    public long getFetchMetric() {
+        return totalStats.fetchMetric.sum();
+    }
+    public long getFetchCurrent() {
+        return totalStats.fetchCurrent.count();
+    }
+    public long getFetchTotal() {
+        return totalStats.fetchTotal.count();
+    }
+    public long getExpandSearchMetric() {
+        return totalStats.expandSearchMetric.sum();
+    }
+    public long getExpandSearchCurrent() {
+        return totalStats.expandSearchCurrent.count();
+    }
+    public long getExpandSearchTotal() {
+        return totalStats.expandSearchTotal.count();
+    }
+    public void incQueryCurrent() { totalStats.queryCurrent.inc(); }
+    public void decQueryCurrent() { totalStats.queryCurrent.dec(); }
+    public void incQueryTotal() { totalStats.queryTotal.inc(); }
+    public void incFetchCurrent() { totalStats.fetchCurrent.inc(); }
+    public void decFetchCurrent() { totalStats.fetchCurrent.dec(); }
+    public void incFetchTotal() { totalStats.fetchTotal.inc(); }
+    public void incExpandSearchCurrent() { totalStats.expandSearchCurrent.inc(); }
+    public void decExpandSearch() { totalStats.expandSearchCurrent.dec(); }
+    public void incExpandSearch() { totalStats.expandSearchTotal.inc(); }
+
+    public void setStats(SearchCoordinatorStats searchCoordinatorStats) {
```

Comment:
Remove setter method from here. We should not allow external parties setting this. As it is created once per node level.
Keep as similar as Shard level stats.